### PR TITLE
LOG-6034: tune audit sources to release file handles

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR modifies audit sources:
* with max read bytes
* ignore files older then one hour
* release rotated files after 5s

### Links
https://issues.redhat.com/browse/LOG-6034
Forward port of #2768

### CODE REVIEW

**Review by alanconway — APPROVED** [review]
/lgtm

**Comment by jcantrill** [comment]
/retest

@alanconway many audit tests are failing.  I'm wondering if its because we are ignoring those older then an hour which could be likely if we do not have lots of activity on a cluster.  Should we remove it?

**Comment by vparfonov** [comment]
@jcantrill i should be [rotate_wait_secs](https://vector.dev/docs/reference/configuration/sources/file/#rotate_wait_secs), see here https://github.com/openshift/cluster-logging-operator/pull/2461

File: internal/generator/vector/conf/complex.toml [hunk 1]
```diff
@@ -48,6 +54,9 @@ type = "file"
 include = ["/var/log/oauth-apiserver/audit.log","/var/log/openshift-apiserver/audit.log","/var/log/oauth-server/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/conf/complex.toml:59]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/conf/complex.toml [hunk 2]
```diff
@@ -63,6 +72,9 @@ type = "file"
 include = ["/var/log/ovn/acl-audit-log.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/conf/complex.toml:77]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/conf/complex_http_receiver.toml [hunk 3]
```diff
@@ -18,6 +18,9 @@ type = "file"
 include = ["/var/log/audit/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/conf/complex_http_receiver.toml:23]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/conf/complex_http_receiver.toml [hunk 4]
```diff
@@ -33,6 +36,9 @@ type = "file"
 include = ["/var/log/kube-apiserver/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/conf/complex_http_receiver.toml:41]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/conf/complex_http_receiver.toml [hunk 5]
```diff
@@ -48,6 +54,9 @@ type = "file"
 include = ["/var/log/oauth-apiserver/audit.log","/var/log/openshift-apiserver/audit.log","/var/log/oauth-server/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/conf/complex_http_receiver.toml:59]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/conf/complex_http_receiver.toml [hunk 6]
```diff
@@ -63,6 +72,9 @@ type = "file"
 include = ["/var/log/ovn/acl-audit-log.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/conf/complex_http_receiver.toml:77]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/input/audit.toml [hunk 7]
```diff
@@ -4,6 +4,9 @@ type = "file"
 include = ["/var/log/audit/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/input/audit.toml:9]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/input/audit.toml [hunk 8]
```diff
@@ -19,6 +22,9 @@ type = "file"
 include = ["/var/log/kube-apiserver/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/input/audit.toml:27]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/input/audit.toml [hunk 9]
```diff
@@ -34,6 +40,9 @@ type = "file"
 include = ["/var/log/oauth-apiserver/audit.log","/var/log/openshift-apiserver/audit.log","/var/log/oauth-server/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/input/audit.toml:45]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/input/audit.toml [hunk 10]
```diff
@@ -49,6 +58,9 @@ type = "file"
 include = ["/var/log/ovn/acl-audit-log.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/input/audit.toml:63]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/input/audit_host.toml [hunk 11]
```diff
@@ -4,6 +4,9 @@ type = "file"
 include = ["/var/log/audit/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/input/audit_host.toml:9]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/input/audit_kube.toml [hunk 12]
```diff
@@ -4,6 +4,9 @@ type = "file"
 include = ["/var/log/kube-apiserver/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/input/audit_kube.toml:9]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/input/audit_openshift.toml [hunk 13]
```diff
@@ -4,6 +4,9 @@ type = "file"
 include = ["/var/log/oauth-apiserver/audit.log","/var/log/openshift-apiserver/audit.log","/var/log/oauth-server/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/input/audit_openshift.toml:9]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/input/audit_ovn.toml [hunk 14]
```diff
@@ -4,6 +4,9 @@ type = "file"
 include = ["/var/log/ovn/acl-audit-log.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/input/audit_ovn.toml:9]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/source/audit_logs.go [hunk 15]
```diff
@@ -12,6 +12,9 @@ type = "file"
 include = ["/var/log/audit/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/source/audit_logs.go:17]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/source/audit_logs.go [hunk 16]
```diff
@@ -24,6 +27,9 @@ type = "file"
 include = ["/var/log/oauth-apiserver/audit.log","/var/log/openshift-apiserver/audit.log","/var/log/oauth-server/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/source/audit_logs.go:32]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/source/audit_logs.go [hunk 17]
```diff
@@ -37,6 +43,9 @@ type = "file"
 include = ["/var/log/kube-apiserver/audit.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/source/audit_logs.go:48]:
```suggestion
rotate_wait_secs = 5
```

File: internal/generator/vector/source/audit_logs.go [hunk 18]
```diff
@@ -50,6 +59,9 @@ type = "file"
 include = ["/var/log/ovn/acl-audit-log.log"]
 host_key = "hostname"
 glob_minimum_cooldown_ms = 15000
+ignore_older_secs = 3600
+max_read_bytes = 3145728
+rotate_wait_ms = 5000
```
Comment (vparfonov) [inline internal/generator/vector/source/audit_logs.go:64]:
```suggestion
rotate_wait_secs = 5
```

**Review by vparfonov — CHANGES_REQUESTED** [review]
```suggestion
rotate_wait_secs = 5
```

**Pull request merged**